apicurio-registry/pkg/controller/apicurioregistry/apicurioregistry_controller.go

```diff
@@ -0,0 +1,196 @@
+			// Owned objects are automatically garbage collected. For additional cleanup logic use finalizers.
+			// Return and don't requeue
+			return reconcile.Result{}, nil
+		}
+		// Error reading the object - requeue the request.
+		return reconcile.Result{}, err
+	}
+
+	// List all pods owned by this PodSet instance
+	lbls := labels.Set{
+		"app": podSet.Name,
+	}
+	existingPods := &corev1.PodList{}
+	err = r.client.List(
+		context.TODO(),
+		&client.ListOptions{
+			Namespace:     request.Namespace,
+			LabelSelector: labels.SelectorFromSet(lbls),
+		},
+		existingPods)
+	if err != nil {
+		reqLogger.Error(err, "failed to list existing pods in the podSet")
+		return reconcile.Result{}, err
+	}
+	existingPodNames := []string{}
+	// Count the pods that are pending or running as available
+	for _, pod := range existingPods.Items {
+		if pod.GetObjectMeta().GetDeletionTimestamp() != nil {
+			continue
+		}
+		if pod.Status.Phase == corev1.PodPending || pod.Status.Phase == corev1.PodRunning {
+			existingPodNames = append(existingPodNames, pod.GetObjectMeta().GetName())
+		}
+	}
+
+	reqLogger.Info("Checking podset", "expected replicas", podSet.Spec.Replicas, "Pod.Names", existingPodNames)
+	// Update the status if necessary
+	status := registryv1alpha1.ApicurioRegistryStatus{
+		Replicas: int32(len(existingPodNames)),
+		PodNames: existingPodNames,
+	}
+	if !reflect.DeepEqual(podSet.Status, status) {
+		podSet.Status = status
+		err := r.client.Status().Update(context.TODO(), podSet)
+		if err != nil {
+			reqLogger.Error(err, "failed to update the podSet")
+			return reconcile.Result{}, err
+		}
+	}
+	// Scale Down Pods
+	if int32(len(existingPodNames)) > podSet.Spec.Replicas {
+		// delete a pod. Just one at a time (this reconciler will be called again afterwards)
+		reqLogger.Info("Deleting a pod in the podset", "expected replicas", podSet.Spec.Replicas, "Pod.Names", existingPodNames)
+		pod := existingPods.Items[0]
+		err = r.client.Delete(context.TODO(), &pod)
+		if err != nil {
+			reqLogger.Error(err, "failed to delete a pod")
+			return reconcile.Result{}, err
+		}
+	}
+
+	// Scale Up Pods
+	if int32(len(existingPodNames)) < podSet.Spec.Replicas {
+		// create a new pod. Just one at a time (this reconciler will be called again afterwards)
+		reqLogger.Info("Adding a pod in the podset", "expected replicas", podSet.Spec.Replicas, "Pod.Names", existingPodNames)
+		pod := newPodForCR(podSet)
+		if err := controllerutil.SetControllerReference(podSet, pod, r.scheme); err != nil {
+			reqLogger.Error(err, "unable to set owner reference on new pod")
+			return reconcile.Result{}, err
+		}
+		err = r.client.Create(context.TODO(), pod)
+		if err != nil {
+			reqLogger.Error(err, "failed to create a pod")
+			return reconcile.Result{}, err
+		}
+	}
+	return reconcile.Result{Requeue: true}, nil
+}
+
+// newPodForCR returns a busybox pod with the same name/namespace as the cr
+func newPodForCR(cr *registryv1alpha1.ApicurioRegistry) *corev1.Pod {
```

Comment:
Why not leverage `Deployment` instead of implementing yourself a kind of replication controller?

From doc
`Deployments are intended to replace Replication Controllers.  They provide the same replication functions (through Replica Sets) and also the ability to rollout changes and roll them back if necessary.`